pkg/injector/bootstrap.go

```diff
@@ -100,3 +105,103 @@ func (wh *mutatingWebhook) marshalAndSaveBootstrap(name, namespace string, confi
 	log.Debug().Msgf("Creating bootstrap config for Envoy: name=%s, namespace=%s", name, namespace)
 	return wh.kubeClient.CoreV1().Secrets(namespace).Create(context.Background(), secret, metav1.CreateOptions{})
 }
+
+// NewBootstrapSecretRotator returns a new bootstrap secret rotator.
+func NewBootstrapSecretRotator(ctx context.Context, kubeClient kubernetes.Interface, informerCollection *informers.InformerCollection, certManager *certificate.Manager, checkInterval time.Duration) *BootstrapSecretRotator {
+	return &BootstrapSecretRotator{
+		context:            ctx,
```

Comment:
from go: https://go.dev/blog/context-and-structs, you should avoid setting contexts on structs. Instead, pass it into the function that requires it